Update Documentation

OmkarPh opened the pull request.

OmkarPh committed:
Added dev script for docs & installation steps for centos

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Merged skeleton docs

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Docs for Tableview

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Added directory-tree docs & column list for tableview

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Updated docs with latest information

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

AyanSinhaMahapatra commented on docs/source/conf.py:
```diff
@@ -18,12 +18,8 @@
-version = '0.0.1'
-# The full version, including alpha/beta/rc tags
-release = version
+copyright = '2023 nexB Inc.'
```
See https://github.com/nexB/skeleton/blob/main/docs/source/conf.py#L20 
We don't want year in the copyright. And others can be included in both copyright and authors.

OmkarPh replied:
I've removed the year from here, and other places too replacing statements like `Copyright (c) 2019 nexB Inc. and others. All rights reserved.` & `Copyright (c) 2017 - 1019 nexB Inc. and others. All rights reserved.`
with `Copyright (c) nexB Inc. and others. All rights reserved.`

Similarly, added `and others` wherever required

AyanSinhaMahapatra commented on docs/requirements.txt:
```diff
@@ -1,26 +1,7 @@
-sphinxcontrib-qthelp==1.0.3
-sphinxcontrib-serializinghtml==1.1.4
-urllib3==1.26.5
+Sphinx>=7.0.0
```
Use these version constraints in the requirement file: https://github.com/nexB/skeleton/blob/main/setup.cfg#L60

OmkarPh replied:
alright
are there some compatibility issues with later versions ?

OmkarPh replied:
Updated

AyanSinhaMahapatra commented on docs/README.md:
````diff
@@ -3,12 +3,16 @@ The first step you want to do is create the python virtual environment:
 ```
 $ cd docs/
 $ python3 -m venv .venv
````
```suggestion
$ python3 -m venv venv
```

This creates a virtualenv named `.venv`, let's have just `venv` instead. And we can either move this to the root of the repo (which is usually done in other projects, but here the venv is just for the docs), or modify our gitignore to include this `docs/venv/` directory so the virtualenv and it's entire contents don't appear in our version control.

OmkarPh replied:
yeah, we don't need venv for entire repo, same for requirements.txt
I'll change venv command

Also, i have added separate .gitignore (python specific) for `docs/` directory other than the root `.gitignore`  (electron specific)
should we keep it like this or have only single .gitignore for entire repo?

OmkarPh replied:
Changed !

AyanSinhaMahapatra commented on docs/Makefile:
```diff
@@ -14,6 +15,13 @@ help:
 .PHONY: help Makefile
 
+# Run the development server using sphinx-autobuild
+dev:
```
What about `make html` (consistent with what we have on the docs elsewhere) or `make docs` instead of `make dev` (which is usually used for development and not docs)

OmkarPh replied:
```
# Catch-all target: route all unknown targets to Sphinx using the new
# "make mode" option.  $(O) is meant as a shortcut for $(SPHINXOPTS).
%: Makefile
	@$(SPHINXBUILD) -M $@ "$(SOURCEDIR)" "$(BUILDDIR)" $(SPHINXOPTS) $(O)
```

This snippet provided by Sphinx itself handles different modes like html, docs, etc

OmkarPh replied:
Changed dev to docs

OmkarPh replied:
the `make html` is already reserved by sphinx, so i don't think we should override this

AyanSinhaMahapatra commented on docs/source/technical-reference/index.rst:
```diff
@@ -19,50 +19,61 @@ ScanCode Workbench Supported Platforms
-+-------------+------------------+---------+------------+-----------------------------------------+
-| Linux RH    |  RHEL 7.xx       |    x64  |     ?      |                                         |
-+-------------+------------------+---------+------------+-----------------------------------------+
++-------------+------------------+----------+------------+-----------------------------------------+
```
Not sure if we've actually tested all these platforms, so let's remove this table and other details and just mention that we support linux/mac/macM1/windows platforms.

OmkarPh replied:
oh, they were already there, i just added the latest versions of macOS, ubuntu, and windows11

but yeah, the latest workbench isn't tested on these specific distros, so it has only `linux/mac/macM1/windows platforms` makes sense

OmkarPh replied:
Updated

AyanSinhaMahapatra reviewed (changes requested):
Thanks++ @OmkarPh 
See initial comments for your consideration.
The autobuild feature works great and is nice to have. Works on linux allright, so let's add it to the skeleton docs too, so it's there in other projects. 
There are some warnings and errors shown on `make dev`:
```
ERROR: Unexpected indentation.
WARNING: Title underline too short
WARNING: duplicate label 
WARNING: Duplicate explicit target name
```
Would make sense to fix these too.

OmkarPh committed:
Removed year from project's copyright statements

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh commented:
> ```
> ERROR: Unexpected indentation.
> WARNING: Title underline too short
> WARNING: duplicate label 
> WARNING: Duplicate explicit target name
> ```

Oops, missed those

Edit: Fixed !

OmkarPh committed:
Fixed sphinx errors, changed requirements versions, formatting and build errors

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Merge branch 'feature/unittests' into update/docs

OmkarPh committed:
Updated documentation scripts

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

AyanSinhaMahapatra commented:
@OmkarPh please add here the .readthedocs.yml file present at https://github.com/nexB/skeleton/blob/main/.readthedocs.yml also

OmkarPh committed:
Updated docs config, dark theme, UI reference updates

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Updated config for readthedocs deploys

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh committed:
Completed UI references, refer-scans, sct-swb interlinks

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh commented:
Hey @AyanSinhaMahapatra 
I've incorporated the changes in docs, as discussed last week

Also, added some suggestions for the scancode-toolkit along with this PR - https://github.com/nexB/scancode-toolkit/pull/3549

Can you push your doc improvements, or maybe merge into this branch itself

AyanSinhaMahapatra reviewed (approved):
@OmkarPh LGTM and thanks!
The docs at https://scancode-workbench.readthedocs.io/en/update-docs/ is now much more comprehensive and looks great.

All the points on more references between sections, using screenshots instead of gifs wherever possible and the misc updates in UI references are addressed and now looking great. :bow:

I'll add my few changes on top and we're ready to merge!

AyanSinhaMahapatra committed:
Add minor fixes in workbench docs

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

AyanSinhaMahapatra reviewed (commented):
@OmkarPh I just noticed we are actually not running the doc check scripts (build and style checks) present at https://github.com/nexB/scancode-workbench/tree/update/docs/docs/scripts in the CI. These have some minor errors to fix too, and added to a github actions. 

See https://github.com/nexB/skeleton/blob/main/.github/workflows/docs-ci.yml 
While you're at it, can you remove the chmod line at skeleton, I thing that's not required, if we chmod and commit the results.

OmkarPh committed:
Added Docs CI, fixed minor docs errors

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

OmkarPh commented:
> @OmkarPh I just noticed we are actually not running the doc check scripts (build and style checks) present at https://github.com/nexB/scancode-workbench/tree/update/docs/docs/scripts in the CI. These have some minor errors to fix too, and added to a github actions.

I've fixed the errors & the new docs CI is working fine now

> See https://github.com/nexB/skeleton/blob/main/.github/workflows/docs-ci.yml While you're at it, can you remove the chmod line at skeleton, I thing that's not required, if we chmod and commit the results.

Yep, chmod is not required. I'll do this in skeleton PR also.

For future reference, tracking chmod using - `git update-index --chmod=+x docs/scripts/doc8_style_check.sh`

OmkarPh commented:
Updated chmod & other docs in skeleton https://github.com/nexB/skeleton/pull/83

AyanSinhaMahapatra committed:
Add minor doc updates

Signed-off-by: Ayan Sinha Mahapatra <ayansmahapatra@gmail.com>

johnmhoran commented:
@OmkarPh @AyanSinhaMahapatra The updated ReadTheDocs looks great.  It's well-organized, the expanations are clear and concise, and there are many good-looking (and helpful) illustrations.

I've also loaded a good-sized scan into SCWB 4.0.0rc4, and if possible I like this version even better than rc3.  I did not realize how many different data views we had until I took the time to explore the data -- nicely done and very useful.  The initial load process was fast and error-free.  Navigation is fast fast fast.

I did notice that in the Table View, the values in the filter dropdowns for `File extension`, `Copyright Holder` and `Copyright Author` were not displayed in alphabetical order (I was unable to ID a pattern to the sort order).  I think the sort order at some point in the past was alphabetical, which imho ought to be the default sort.

DennisClark reviewed (approved):
Everything looks good to me.  The documentation is excellent and quite thorough.

OmkarPh commented:
> I did notice that in the Table View, the values in the filter dropdowns for `File extension`, `Copyright Holder` and `Copyright Author` were not displayed in alphabetical order (I was unable to ID a pattern to the sort order). I think the sort order at some point in the past was alphabetical, which imho ought to be the default sort.

yeah, having an alphabetical order would be nice, I'll update that in a different PR

OmkarPh commented:
Thanks for the review @johnmhoran @DennisClark @AyanSinhaMahapatra 😄

OmkarPh committed:
Updated underlying technology, build tools versions, removed policies outdated docs

Signed-off-by: Omkar Phansopkar <omkarphansopkar@gmail.com>

AyanSinhaMahapatra commented:
Thanks @OmkarPh for the updates, ready at last, merging!
